testjson/format.go

```diff
@@ -73,6 +75,55 @@ func standardJSONFormat(out io.Writer) EventFormatter {
 	})
 }
 
+func testDoxFormat(out io.Writer, opts FormatOptions) EventFormatter {
+	buf := bufio.NewWriter(out)
+	type Result struct {
+		Event    TestEvent
+		Sentence string
+	}
+	getIcon := icon
+	if opts.UseHiVisibilityIcons {
+		getIcon = iconHiVis
+	}
+	results := map[string][]Result{}
+	return eventFormatterFunc(func(event TestEvent, exec *Execution) error {
+		switch {
+		case event.PackageEvent():
+			if !event.Action.IsTerminal() {
+				return nil
+			}
+			if len(results[event.Package]) == 0 {
+				// No testdox for you
+				return nil
+			}
+			fmt.Fprintf(buf, "%s:\n", event.Package)
+			tests := results[event.Package]
+			sort.Slice(tests, func(i, j int) bool {
+				return tests[i].Sentence < tests[j].Sentence
+			})
+			for _, r := range tests {
+				fmt.Fprintf(buf, " %s %s %s\n",
+					getIcon(r.Event.Action),
+					r.Sentence,
+					r.Event.ElapsedFormatted())
+			}
+			fmt.Fprintln(buf)
+			return buf.Flush()
+		case event.Action.IsTerminal():
+			// Fuzz test cases tend not to have interesting names,
+			// so only report these if they're failures
+			if strings.HasPrefix(event.Test, "Fuzz") && event.Action == ActionPass {
+				return nil
+			}
```

Comment:
What do you think about keeping these in instead of hiding them? I've written a few fuzz tests that had reasonable names, and I suspect most readers would rather see the bad name and improve it, rather than it being hidden from the output entirely.

Reply:
Just to be 100% sure we're talking about the same thing, what `gotestdox` omits are the names of fuzz _cases_ (which are autogenerated, though of course you can change them). So for me, for example, here are some test names that `gotestdox` would otherwise have printed:

```
 ✔ [fuzz] Prettify (0.01s)
 ✔ [fuzz] Prettify 0 2d 4bc 9f 4e 56704 8ebe 71 9db 6d 9 4d 2 2ed 078 8a 5 6e 3bdf 4aa 1fa 3a 54 6b 5a 68 1a 4 (0.00s)
 ✔ [fuzz] Prettify 0 5bda 4c 4 3f 060615 2eb 866 1c 3f 0 7af 3 5c 2 3ea 15 1e 4c 0eba 1361 6e 68 1baa 0b 7 6b (0.00s)
 ✔ [fuzz] Prettify 1d 9c 42 8aec 8af 9b 4c 3d 6f 8f 8 7bb 3fd 4cd 8877 6a 651 8bf 23 2ca 8ece 0 5de 18 0c 4f (0.00s)
 ✔ [fuzz] Prettify 275 9ad 3 3bc 8 5f 33519 9a 67 8f 5b 2afd 24576 0bc 67 0e 0 3abcfedd 2244026 2e 7f 20 (0.00s)
 ✔ [fuzz] Prettify 38382467 5cc 2c 17 7e 19 8f 7 4ede 3c 8ca 2 9be 0 0e 2b 679 7a 03 4a 7a 583 9a 4cd 9af 1c (0.00s)
 ✔ [fuzz] Prettify 6 8ed 40 3ba 2cdc 6 1b 14331331 2fd 6 7edbd 7e 8bc 648157 0c 7d 1d 7b 690179316 6ce (0.00s)
 ✔ [fuzz] Prettify 728774926 2c 76419 8e 0bfd 31301520 2b 150 3cb 01 1ed 5 8da 51385 3e 0fb 2e 68601 (0.00s)
 ✔ [fuzz] Prettify 93 1b 6 1b 89 5a 3 7b 9f 6d 19 6cc 7312 3abb 8 3cdd 7bc 0 5e 46 9ad 7d 4b 7 6bd 1ff 1 6db 60 (0.00s)
 ✔ [fuzz] Prettify 9eca 0c 1 5d 9a 0 7d 782 1d 4 3cbb 4e 0 8d 4b 8 4ca 61 0d 67 9b 1ee 8 9a 0f 0a 9cdf 4 6f 5b 79 (0.00s)
 ✔ [fuzz] Prettify aa 1 2d 4e 416 6bf 7ad 0 0d 5 9c 3ec 5e 9ef 47349 4bd 2fde 7d 9 3b 4813 6abbc 7 7a 5acef (0.00s)
 ✔ [fuzz] Prettify d 7258 6fff 5c 01205 (0.00s)
 ✔ [fuzz] Prettify f 5d 2 6b 648617975625 8edcb 8fde 2 3e 65 5aeae 6f 7ce 1d 0d 2238060 9e 505 5cd 6de (0.00s)
 ✔ [fuzz] Prettify seed# 0 (0.00s)
 ✔ [fuzz] Prettify seed# 1 (0.00s)
 ✔ [fuzz] Prettify seed# 11 (0.00s)
 ✔ [fuzz] Prettify seed# 12 (0.00s)
 ✔ [fuzz] Prettify seed# 13 (0.00s)
 ✔ [fuzz] Prettify seed# 14 (0.00s)
 ✔ [fuzz] Prettify seed# 15 (0.00s)
 ✔ [fuzz] Prettify seed# 16 (0.00s)
 ✔ [fuzz] Prettify seed# 17 (0.00s)
 ✔ [fuzz] Prettify seed# 18 (0.00s)
 ✔ [fuzz] Prettify seed# 19 (0.00s)
 ✔ [fuzz] Prettify seed# 2 (0.00s)
 ✔ [fuzz] Prettify seed# 20 (0.00s)
 ✔ [fuzz] Prettify seed# 21 (0.00s)
 ✔ [fuzz] Prettify seed# 22 (0.00s)
 ✔ [fuzz] Prettify seed# 23 (0.00s)
 ✔ [fuzz] Prettify seed# 24 (0.00s)
 ✔ [fuzz] Prettify seed# 25 (0.00s)
 ✔ [fuzz] Prettify seed# 26 (0.00s)
 ✔ [fuzz] Prettify seed# 27 (0.00s)
 ✔ [fuzz] Prettify seed# 28 (0.00s)
 ✔ [fuzz] Prettify seed# 29 (0.00s)
 ✔ [fuzz] Prettify seed# 3 (0.00s)
 ✔ [fuzz] Prettify seed# 30 (0.00s)
 ✔ [fuzz] Prettify seed# 31 (0.00s)
 ✔ [fuzz] Prettify seed# 32 (0.00s)
 ✔ [fuzz] Prettify seed# 33 (0.00s)
 ✔ [fuzz] Prettify seed# 34 (0.00s)
 ✔ [fuzz] Prettify seed# 35 (0.00s)
 ✔ [fuzz] Prettify seed# 36 (0.00s)
 ✔ [fuzz] Prettify seed# 37 (0.00s)
 ✔ [fuzz] Prettify seed# 38 (0.00s)
 ✔ [fuzz] Prettify seed# 4 (0.00s)
 ✔ [fuzz] Prettify seed# 5 (0.00s)
 ✔ [fuzz] Prettify seed# 6 (0.00s)
 ✔ [fuzz] Prettify seed# 7 (0.00s)
 ✔ [fuzz] Prettify seed# 8 (0.00s)
 ✔ [fuzz] Prettify seed# 9 (0.00s)
 ```

It makes complete sense for `gotestsum` to print these, of course, since they're tests! However, a documentation generator can't make anything useful of them.